R/getPrevalence.R

```diff
@@ -311,6 +333,63 @@ setMethod("getRareTaxa", signature = c(x = "SummarizedExperiment"),
     }
 )
 
+############################# subsetByPrevalentTaxa ############################
+
+#' @rdname getPrevalence
+#' @export
+setGeneric("subsetByPrevalentTaxa", signature = "x",
+           function(x, ...)
+               standardGeneric("subsetByPrevalentTaxa"))
+
+#' @rdname getPrevalence
+#' @export
+setMethod("subsetByPrevalentTaxa", signature = c(x = "SummarizedExperiment"),
+    function(x, rank = NULL, ...){
+        # If rank is not NULL
+        if( !is.null(rank) ){
+            # Checks rank
+            .check_taxonomic_rank(rank, x)
+            # Agglomerates the object by rank. Taxa that do not have information
+            # at specific rank are excluded.
+            x <- agglomerateByRank(x, rank = rank, na.rm = TRUE)
+            # Changes rank to NULL, so that agglomeration is not done in next steps
+            rank <- NULL
+        }
+        # Gets the prevalent taxa
+        prevalent_taxa <- getPrevalentTaxa(x, ...)
+        # Subsets the object based on prevalent taxa
+        x <- x[prevalent_taxa]
```

Comment:
This works just by accident. Since an SE is a two dimensional object, you should specify the dimension explicitly.

```r
x[prevalent_taxa,]
```

btw: What happens is rownames are empty? There needs to be an alternative path for that case. This also needs to be fixed for the other three functions:

```r
> data(esophagus)
> getPrevalentTaxa(esophagus)
 [1] "59_8_22" "59_8_12" "65_3_22" "65_7_12" "65_2_17" "65_7_4"  "65_1_17" "65_8_7"  "59_5_2"  "65_6_2"  "65_4_10" "9_7_25"  "9_4_5"  
[14] "65_1_8"  "65_2_5"  "59_7_6"  "59_9_31" "59_9_18" "65_9_13" "65_3_18" "65_8_25" "59_9_17" "9_4_6"   "59_4_25" "59_5_19" "65_9_9" 
[27] "59_2_6" 
> rownames(esophagus) <- NULL
> getPrevalentTaxa(esophagus)
NULL
```